Added watch for Sentinel message +switch-master

McMlok opened the pull request:
Hello,

I have same situation as you. On-Premise Redis with sentinels for fail over. When I use your solution for ASP.NET Cache I can connect to actual master. But when this master crash down sentinel wasn't connected for address of new master. I added two catches to retry method and subscription to sentinel message which was published when master was changed. Can you look at it?

Regards
McMlok

DamirAinullin committed:
Fix issue #2 from Github: Mono: NullReferenceException for concurrent requests.

dvinales committed:
Using hash tags to support Redis Cluster

change the key generator class to support Redis Cluster according to the
specs (http://redis.io/topics/cluster-spec)

SiddharthChatrolaMs committed:
Merge pull request #5 from dvinales/master

Using hash tags to support Redis Cluster

Commit:
Information about change in Readme that pop-ups when package installed

Commit:
Readme comment modified

Commit:
Added Release Notes link in readme

Commit:
Add link in readme that opens up on installation

Commit:
Update SE.Redis version to latest

Commit:
Updating SE.Redis in projects

Commit:
Update blog to wiki link for release notes

Commit:
Removing CommonAssemblyInfo so we can create seperate release for output cache and session state

McMlok committed:
Added watch for Sentinel message +switch-master and catch for connection and readonly excpetions

cjb5790 commented:
@McMlok I can certainly take a look. I have some additional work to do on this pull request before it will be accepted.

McMlok committed:
Looking if sentinel is available if connection to sentinell fails connection is reestablished to another sentinell.

Commit:
If customer is using connection string then also set "abortConnect=False" if abortConnect is not provided inside connection string

Commit:
Always set abortconnect to false

Commit:
Update StackExchange.Redis.StrongName to 1.0.488

Commit:
Output cache functional test should use different data

andreymir committed:
Allow session provider to use connection string from <connectionStrings> section

andreymir committed:
Allow session provider to use connection string from <connectionStrings> section: unit test

robertmiles3 committed:
Specify initial list size for performance

Since the count of session keys is known up front, we can save internal
List<T> re-allocations by specifying initial size up front.

Commit:
Add xunit.runner.visualstudio in test projects

SiddharthChatrolaMs committed:
Merge pull request #23 from andreymir/issue-17

use connection string from <connectionStrings> section

Commit:
Read connectionString from customer given static method

Commit:
Updated test and fixed small bug

Commit:
Trying to improve flaky tests

Commit:
Throw if both connection string and settings are provided to fetch connection

Commit:
Updated ConnectionStringException string

Commit:
typo fixed

Commit:
Update top comment in web.config section

Commit:
Added wiki link in web.config comment

Commit:
Fixed broken test

brendankowitz committed:
Fixes lua script where PEXPIRE call will fail for negative values which creates an output cache key in redis with no expiry (-1)

Commit:
Added Code of Conduct for Microsoft open source projects

SiddharthChatrolaMs committed:
Merge pull request #31 from brendankowitz/improve-add-key-lua

Fixes lua script where PEXPIRE call will fail for negative values which creates an output cache key in redis with no expiry (-1)

SiddharthChatrolaMs committed:
Merge pull request #26 from robertmiles3/master

Specify initial list size for performance

SiddharthChatrolaMs committed:
Merge pull request #3 from DamirAinullin/master

Fix issue #2 from Github: Mono: NullReferenceException for concurrent…

Commit:
Extensible mechanism to provide different serialization method

Commit:
Refactor RedisUtility to allow providing a different serialization method

Commit:
Similar to InProc session should use first time used key in case if "casing" changes

Commit:
Revert "Fixes lua script where PEXPIRE call will fail for negative values which creates an output cache key in redis with no expiry (-1)"

- GET directly is better way then checking TTL and then doing GET as it may result into race condition
- Using PSETEX instead of combination of SET and PEXPIRE so we don't have key with no expiry

Commit:
Using ToUpperInvariant instead of ToLower

SiddharthChatrolaMs committed:
Merge pull request #38 from thepirat000/master

Extensible mechanism to provide different serialization method

Commit:
Create RedisUtility only once and share across all requests. Added test using custom Serializer

Commit:
Add comment in web.config for redisSerializerType parameter

steved0x committed:
Redis Cache - updated links to Session State Provider and Output Cache provider documents on Azure.com

SiddharthChatrolaMs committed:
Merge pull request #43 from steved0x/patch-1

Redis Cache - updated links to Session State Provider and Output Cach…

teonivalois committed:
Updating redis client library

Commit:
Fixed the issue when databaseId from connection string ignored

SiddharthChatrolaMs committed:
Merge pull request #53 from Almazen/master

Fixed the issue when databaseId from connection string ignored

SiddharthChatrolaMs committed:
Merge pull request #47 from teonivalois/master

Updating redis client library

Commit:
Update NuGet.Config

Commit:
Using msbuild version 14 VS 2015 and updating redis-64 NuGet for test

Commit:
Moving to xunit 2.*

Commit:
Test projects moved to .net 4.5 so no need of BCL packages

Commit:
Update StackExchange.Redis.StrongName version in nuspec

Commit:
desirialization session data only when needed

Commit:
All Enumeration related code should work fine after lazy deserialization

Commit:
Rewrite some part of ChangeTrackingSessionStateItemCollection to reduce different dictionaries

Commit:
Other small code review chnages

Commit:
Code review comments fixed

Commit:
Simple design changes

Commit:
Method reduction

Commit:
Added test case for actual null value

Commit:
Fixing a typo in comment

Commit:
Update StackExchange.Redis to latest version

McMlok committed:
merge
